### PR TITLE
[Teaser] Add Pretitle text

### PR DESCRIPTION
**Pull request by dymitrs**
| Q                        | A 
| ------------------------ | ---
| Patch: Bug Fix?          | No
| Minor: New Feature?      | Pretitle in Teaser's dialog
| Major: Breaking Change?  | No
| Tests Added + Pass?      | Yes
| Documentation Provided   | Yes 
| Any Dependency Changes?  | No
| License                  | Apache License, Version 2.0

Added pretitle field in Teaser dialog. Added tests & updated README.
Fixing #823

### CODE REVIEW

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/Teaser.java [hunk 1]
```diff
@@ -164,6 +171,16 @@ default boolean isImageLinkHidden() {
         throw new UnsupportedOperationException();
     }
 
+    /**
+     * Returns this preteaser's title, if one was defined.
```
Comment (richardhand) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/Teaser.java]: @dymitrs - "Returns this teaser's pretitle, if one was defined"

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/Teaser.java [hunk 2]
```diff
@@ -164,6 +171,16 @@ default boolean isImageLinkHidden() {
...
+     *
+     * @return the teaser's title or {@code null}
```
Comment (richardhand) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/Teaser.java]: `@return the teaser's pretitle or {@code null}`

File: content/src/content/jcr_root/apps/core/wcm/components/teaser/v1/teaser/_cq_dialog/.content.xml [hunk 3]
```diff
@@ -67,6 +67,19 @@
                                         jcr:primaryType="nt:unstructured"
                                         sling:resourceType="granite/ui/components/coral/foundation/container">
                                         <items jcr:primaryType="nt:unstructured">
+                                            <pretitleGroup
```
Comment (richardhand) [inline content/src/content/jcr_root/apps/core/wcm/components/teaser/v1/teaser/_cq_dialog/.content.xml]: @dymitrs - I don't think this needs to be grouped like the title or description, would remove the well wrapper here.

File: content/src/content/jcr_root/apps/core/wcm/components/teaser/v1/teaser/pretitle.html [hunk 4]
```diff
@@ -0,0 +1,18 @@
+<!--/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  ~ Copyright 2018 Adobe
```
Comment (richardhand) [inline content/src/content/jcr_root/apps/core/wcm/components/teaser/v1/teaser/pretitle.html]: copyright year 2019

**Comment by richardhand** [comment]
@dymitrs we will have to fix the merge conflicts here as there was a recent restructuring of the component library to `ui.apps` and `ui.content` in https://github.com/adobe/aem-core-wcm-components/commit/68c4071bd6f94de711510d07a8eadd408b2f0dd6

File: content/src/content/jcr_root/apps/core/wcm/components/teaser/v1/teaser/_cq_dialog/.content.xml (hunk 3 above)
Comment (dymitrs) [reply content/src/content/jcr_root/apps/core/wcm/components/teaser/v1/teaser/_cq_dialog/.content.xml]: Sure, I'll remove it

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/TeaserImpl.java [hunk 5]
```diff
@@ -126,6 +128,11 @@ private void initModel() {
             targetPage = pageManager.getPage(linkURL);
         }
 
+        if (pretitleHidden) {
+            pretitle = null;
```
Comment (jckautzmann) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/TeaserImpl.java]: I would not change the value of the `pretitle` if it is hidden: the value should be the one defined by the author, even if the field is hidden.

Comment (richardhand) [reply bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/TeaserImpl.java]: I agree, but we already do the same for the title and description and `pretitleHidden` isn't exposed to the template so the only way to check for display is `? teaser.pretitle`. Perhaps it's best to be consistent here with the current behaviour?

Comment (jckautzmann) [reply bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/TeaserImpl.java]: hm... I agree that we should be consistent and have the same behavior as for `title` and `description`.

Comment (dymitrs) [reply bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/TeaserImpl.java]: Yes, I tried to be consistent and have the same behavior as for the other fields.

**Pull request merged**